src/core/load-fragment.c

```diff
@@ -4730,26 +4730,47 @@ int config_parse_exec_directories(
                 /* For State and Runtime directories we support an optional destination parameter, which
                  * will be used to create a symlink to the source. */
                 _cleanup_free_ char *dresolved = NULL;
-                if (!isempty(dest)) {
+                ExecDirectoryFlags exec_directory_flags = 0;
+                if (!isempty(dest_or_flags)) {
+                        const char *destination = NULL;
+
                         if (streq(lvalue, "ConfigurationDirectory")) {
                                 log_syntax(unit, LOG_WARNING, filename, line, 0,
-                                           "Destination parameter is not supported for ConfigurationDirectory, ignoring: %s", tuple);
+                                           "Additional parameter is not supported for ConfigurationDirectory, ignoring: %s", tuple);
                                 continue;
                         }
 
-                        r = unit_path_printf(u, dest, &dresolved);
-                        if (r < 0) {
-                                log_syntax(unit, LOG_WARNING, filename, line, r,
-                                        "Failed to resolve unit specifiers in \"%s\", ignoring: %m", dest);
-                                continue;
+                        if (!isempty(flags)) {
+                                if (!streq(flags, "ro")) {
```

Comment:
instead of having a manual parser here, please add the usual exec_directory_flags_to_string() and exec_directory_flags_from_string() please